mars/dataframe/base/bloom_filter.py

```diff
@@ -0,0 +1,257 @@
+        return bloom_filter
+
+    @classmethod
+    def _convert_to_hashable_dtypes(cls, dtypes: pd.Series):
+        dtypes = dict(
+            (name, dtype) if np.issubdtype(dtype, int) else (name, str)
+            for name, dtype in dtypes.iteritems()
+        )
+        return dtypes
+
+    @classmethod
+    def execute(cls, ctx: Union[dict, Context], op: "DataFrameBloomFilter"):
+        if op.execution_stage == "build":
+            on = op.on
+            in_data = ctx[op.inputs[0].key]
+            if cls._filter_on_index(on, in_data):
+                bloom_filter = cls._build_index_filter(in_data, op)
+            elif isinstance(on, str):
+                bloom_filter = cls._build_series_filter(in_data, op)
+            else:
+                bloom_filter = cls._build_dataframe_filter(in_data, op)
+            ctx[op.outputs[0].key] = bloom_filter
+        elif op.execution_stage == "union":
+            # union bloom filters
+            filters = [ctx[inp.key] for inp in op.inputs]
+            out = filters[0]
+            for f in filters[1:]:
+                out.union(f)
+            ctx[op.outputs[0].key] = out
+        elif op.execution_stage == "filter":
+            on = op.on
+            in_data = ctx[op.inputs[0].key]
+            bloom_filter = ctx[op.inputs[1].key]
+            if cls._filter_on_index(on, in_data):
+                if isinstance(in_data.index, pd.MultiIndex):
+                    idx = in_data.index.names.index(on)
+                    ctx[op.outputs[0].key] = in_data[
+                        in_data.index.map(lambda x: x[idx] in bloom_filter)
+                    ]
+                else:
+                    ctx[op.outputs[0].key] = in_data[
+                        in_data.index.map(lambda x: x in bloom_filter)
+                    ]
+            else:
+                row_func = lambda row: cls._get_value(row) in bloom_filter
+                if isinstance(on, str):
+                    # series
+                    try:
+                        filtered = in_data[in_data[on].map(row_func)]
+                    except TypeError:
+                        converted_data = in_data.astype(str)
+                        filtered = in_data[converted_data[on].map(row_func)]
+                    ctx[op.outputs[0].key] = filtered
+                else:
+                    # dataframe
+                    try:
+                        filtered = in_data[in_data[on].apply(row_func, axis=1)]
+                    except TypeError:
+                        converted_data = in_data.astype(
+                            cls._convert_to_hashable_dtypes(in_data.dtypes)
+                        )
+                        filtered = in_data[converted_data[on].apply(row_func, axis=1)]
+                    ctx[op.outputs[0].key] = filtered
+
+        else:  # pragma: no cover
+            raise ValueError(f"Unknown execution stage: {op.execution_stage}")
+
+
+def filter_by_bloom_filter(
```

Comment:
Maybe this function could be tileable level, and the chunks related stuff could be moved to the `tile` function, I guess this function could be exposed to user some day perhaps, not just be taken as a utility function.